### DISCUSSION TITLE
Smart Meter Interface / calculations

### DISCUSSION THREAD

**Discussion by nagyrobi** [arendst/Tasmota#11133] [Q&A] [answered]
Trying to calculate actual current form the power and voltage values on a [Sanxing SX631](https://tasmota.github.io/docs/Smart-Meter-Interface/#sanxing-sx6x1-sxxu1x-ascii-obis) using [Special Commands](https://tasmota.github.io/docs/Smart-Meter-Interface/#special-commands).

The problem is that voltage is given in V (volts) and power is given in kW (kilowatts) so unit conversion by 1000 is needed to obtain current in A (ampers). This doesn't work with Special Commands.

The formula we all learned in middle school physics class is W = V x A so we can get A = W / V. In this specific case we would either need A = kW * 1000 / V or A = kW / V / 1000.

Translated to SML, this would look like below:
Input data:
```
1,1-0:32.7.0(@1,L1 Voltage,V,volt_l1,1
1,1-0:1.7.0(@1,Power,kW,pwr_in,3
```
These are 231.5 V and  2.147 kW respectively.

Result is 0.009 A while it should be 9,274 A.

Output version1:
`1,=m2/#1000/1@1,Current L1,A,curr_l1,3` ignores `/#1000`

Output version2:
`1,=m2/1*#1000@1,Current L1,A,curr_l1,3` ignores `*#1000`

Output version3:
`1,=m2/1@1000,Current L1,A,curr_l1,3` ignores multiplier after `@` (probably incorrect anyway)

Output version4:
`1,=m2/1@0.001,Current L1,A,curr_l1,3`  ignores multiplier after `@`

What am I doing wrong?

**Comment by gemu2015** [comment, marked as answer]
i just checked the math function and it mostly works as expected. BUT
indeed the scaling factor is not used here
and immediate numbers in this math option must be positive integers and LOWER then max decode lines
both fixed in my fork (please test)

**Reply by nagyrobi** [reply]
Output version4 works now, thank you:
`1,=m2/1@0.001,Current L1,A,curr_l1,3`
